sound/soc/codecs/max98363.c

```diff
@@ -415,6 +414,8 @@ static int max98363_init(struct sdw_slave *slave, struct regmap *regmap)
 	max98363->regmap = regmap;
 	max98363->slave = slave;
 
+	regcache_cache_only(max98363->regmap, true);
```

Comment:
@plbossart I am thinking about regcache_mark_dirty(). We should call regcache_mark_dirty when reg cache is not reliable. If someone write the reg cache when cache only = true, the cache value may not be the same as the real register value and we should call regcache_mark_dirty in this scenario. Maybe it is better to add regcache_mark_dirty(max98363->regmap); after regcache_cache_only(max98363->regmap, true);?
What do you think? @shumingfan

Reply:
@bardliao I think the write operation will set the cache_dirty flag when the cache_only was enabled.
https://github.com/thesofproject/linux/blob/90c8074e2cea7cb56fa2f7902646e3aa5aacf099/drivers/base/regmap/regmap.c#L2004